rustc-demangle-capi/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+#![no_std]
+#![feature(lang_items, core_intrinsics)]
+
+#[cfg(test)]
+#[macro_use]
+extern crate std;
+
+extern crate rustc_demangle;
+
+use core::fmt::Write;
+
+struct WriteAdaptor<'a> {
+    inner: &'a mut [u8]
+}
+
+impl<'a> core::fmt::Write for WriteAdaptor<'a> {
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        if s.len() > self.inner.len() {
+            return Err(core::fmt::Error);
+        }
+        let (left,right) = core::mem::replace(self, WriteAdaptor { inner: &mut [] }).inner.split_at_mut(s.len());
+        left.copy_from_slice(s.as_bytes());
+        *self = WriteAdaptor { inner: right };
+        Ok(())
+    }
+}
+
+/// C-style interface for demangling.
+/// Demangles symbol given in `mangled` argument into `out` buffer
+///
+/// Unsafe as it handles buffers by raw pointers.
+///
+/// Returns 0 if `mangled` is not Rust symbol or if `out` buffer is too small
+/// Returns 1 otherwise
+#[no_mangle]
+pub unsafe extern fn rustc_demangle(mangled: *const u8, out: *mut u8, out_size: usize) -> i64 {
```

Comment:
I think this should return `i32`, right? Or maybe link to the `libc` crate and return `c_int`?

Reply:
With std I was using `use std::os::raw::{c_char,c_int};`. But you are right, it probably should be i32 instead of i64, or preferably c_int if I can get it somewhere easily. I suppose I could add libc as a dependency, although it feels bit heavy solution for getting few newtypes. Of course using std would solve this issue.